### PR TITLE
Parser for iosxe show ap config general

### PR DESCRIPTION
**Pull request by qduk**
## Description
Parser for iosxe show ap config general

## Screenshots:
Capture

## Checklist:
- [X ] I have updated the changelog.
- [ ] I have updated the documentation (If applicable).
- [ X] I have added tests to cover my changes (If applicable).
- [X ] All new and existing tests passed.
- [X ] All new code passed compilation.

### CODE REVIEW

**Review by ThomasJRyan — APPROVED** [review]

**Review by ahersi2-cisco — APPROVED** [review]

File: src/genie/libs/parser/iosxe/show_ap.py [hunk 1]
```diff
@@ -983,3 +983,1443 @@ def cli(self, output=None):
                 continue
 
         return ap_led_brightness_level_summary_dict
+
+# =============================
+# Schema for:
+#  * 'show_ap_config_general'
```
Comment (tahigash) [inline src/genie/libs/parser/iosxe/show_ap.py:989]:
```suggestion
#  * 'show ap config general'
```

File: src/genie/libs/parser/iosxe/show_ap.py [hunk 2]
```diff
@@ -983,3 +983,1443 @@ def cli(self, output=None):
...
+# =============================
+class ShowApConfigGeneralSchema(MetaParser):
+    """Schema for show_ap_config_general."""
```
Comment (tahigash) [inline src/genie/libs/parser/iosxe/show_ap.py:992]:
```suggestion
    """Schema for show ap config general"""
```

File: src/genie/libs/parser/iosxe/tests/ShowApConfigGeneral/cli/equal/golden_output_expected.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
+expected_output = {
+    "ap_config_general_info": {
```
Comment (tahigash) [inline src/genie/libs/parser/iosxe/tests/ShowApConfigGeneral/cli/equal/golden_output_expected.py:2]: `ap_name` is better for the value `bg-1-cap1`.
```
"ap_name": {
    "bg-1-cap1": {

File: src/genie/libs/parser/iosxe/tests/ShowApConfigGeneral/cli/equal/golden_output_expected.py [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+        "bg-1-cap1": {
+            "cisco_ap_identifier": "70b3.d278.e03e",
+            "country_code": "IN",
+            "regulatory_domain_allowed_by_country": "802.11bg:-A   802.11a:-DN",
+            "ap_country_code": "IN  - India",
+            "slot_0": "-A",
+            "slot_1": "-D",
```
Comment (tahigash) [inline src/genie/libs/parser/iosxe/tests/ShowApConfigGeneral/cli/equal/golden_output_expected.py:9]: respect an indentation in output. so, schema could be:
```
"ap_regulatory_domain": {
    "slot_0": "-A",
    "slot_1": "-D",

File: src/genie/libs/parser/iosxe/tests/ShowApConfigGeneral/cli/equal/golden_output_expected.py [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+            "ip_address_configuration": "DHCP",
+            "ip_address": "10.10.5.14",
+            "ip_netmask": "255.255.254.0",
+            "gateway_ip_address": "10.10.5.1",
+            "capwap_path_mtu": 1485,
+            "capwap_active_window_size": 1,
+            "telnet_state": "Disabled",
+            "cpu_type": "ARMv7 Processor rev 1 (v7l)",
+            "memory_type": "DDR3",
+            "memory_size": "1028096 KB",
```
Comment (tahigash) [inline src/genie/libs/parser/iosxe/tests/ShowApConfigGeneral/cli/equal/golden_output_expected.py:20]: It seems the unit is always `KB`. let's have the unit "KB" in key name and value is just number.
```
"memory_size_kb": 1028096,

**Review by tahigash — CHANGES_REQUESTED** [review]

File: src/genie/libs/parser/iosxe/show_ap.py (hunk 1 above)
Comment (qduk) [reply src/genie/libs/parser/iosxe/show_ap.py]: Updated

File: src/genie/libs/parser/iosxe/tests/ShowApConfigGeneral/cli/equal/golden_output_expected.py (hunk 3 above)
Comment (qduk) [reply src/genie/libs/parser/iosxe/tests/ShowApConfigGeneral/cli/equal/golden_output_expected.py]: Updated

File: src/genie/libs/parser/iosxe/show_ap.py (hunk 2 above)
Comment (qduk) [reply src/genie/libs/parser/iosxe/show_ap.py]: Updated

File: src/genie/libs/parser/iosxe/tests/ShowApConfigGeneral/cli/equal/golden_output_expected.py (hunk 4 above)
Comment (qduk) [reply src/genie/libs/parser/iosxe/tests/ShowApConfigGeneral/cli/equal/golden_output_expected.py]: Updated

File: src/genie/libs/parser/iosxe/tests/ShowApConfigGeneral/cli/equal/golden_output_expected.py (hunk 5 above)
Comment (qduk) [reply src/genie/libs/parser/iosxe/tests/ShowApConfigGeneral/cli/equal/golden_output_expected.py]: Updated

**Comment by qduk** [comment]
Updated unit test ran.
ap_general

**Review by tahigash — APPROVED** [review]

**Pull request merged**